src/payment/strategies/adyenv2/adyenv2-payment-strategy.ts

```diff
@@ -0,0 +1,289 @@
+                    return this._store.dispatch(this._paymentActionCreator.submitPayment({...payment, paymentData}));
+                }
+
+                const paymentPayload = {
+                    methodId: payment.methodId,
+                    paymentData: {
+                        nonce: this._getStateContainer(),
+                        shouldSaveInstrument,
+                    },
+                };
+
+                return this._store.dispatch(this._paymentActionCreator.submitPayment(paymentPayload));
+            })
+            .catch(error => {
+                if (!(error instanceof RequestError) || !some(error.body.errors, { code: 'three_d_secure_required' })) {
+                    return Promise.reject(error);
+                }
+
+                if (error.body.three_ds_result.result_code === ResultCode.IdentifyShopper) {
+
+                    return this._handle3DS2Fingerprint(error.body.three_ds_result, payment.methodId)
+                        .then((payment: Payment) =>
+                            this._store.dispatch(this._paymentActionCreator.submitPayment(payment)))
+                        .catch(error => {
+                            if (!(error instanceof RequestError) || !some(error.body.errors, { code: 'three_d_secure_required' })) {
+                                return Promise.reject(error);
+                            }
+
+                            if (error.body.three_ds_result.result_code === ResultCode.ChallengeShopper) {
+                                return this._handle3DS2Challenge(error.body.three_ds_result, payment.methodId)
+                                    .then((payment: Payment) =>
+                                        this._store.dispatch(this._paymentActionCreator.submitPayment(payment)));
+                            }
+
+                            return Promise.reject(error);
+                        });
+                }
+
+                if (error.body.three_ds_result.result_code === ResultCode.ChallengeShopper) {
+
+                    return this._handle3DS2Challenge(error.body.three_ds_result, payment.methodId)
+                        .then((payment: Payment) =>
+                            this._store.dispatch(this._paymentActionCreator.submitPayment(payment)));
+                }
+
+                return new Promise(() => {
+                    this._formPoster.postForm(error.body.three_ds_result.acs_url, {
+                        PaReq: error.body.three_ds_result.payer_auth_request,
+                        TermUrl: error.body.three_ds_result.callback_url,
+                        MD: error.body.three_ds_result.merchant_data,
+                    });
+                });
+            });
+    }
+
+    finalize(): Promise<InternalCheckoutSelectors> {
+        return Promise.reject(new OrderFinalizationNotRequiredError());
+    }
+
+    deinitialize(): Promise<InternalCheckoutSelectors> {
+        if (this._adyenOldComponent) {
```

Comment:
Please see my previous comment about this. You only need to keep the `adyenComponent.unmount` part.

Reply:
I already post my reason to keep old components